test/integration/issue-2155/test.tap.mjs

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2024 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import tap from 'tap'
+import crypto from 'crypto'
+import path from 'path'
+import url from 'url'
+import helper from '../../lib/agent_helper.js'
+import shimmer from '../../../lib/shimmer.js'
+import InstrumentationDescriptor from '../../../lib/instrumentation-descriptor.js'
+
+let modPath
+if (import.meta.dirname) {
+  modPath = path.join(import.meta.dirname, 'foo.cjs')
+} else {
+  modPath = path.join(path.dirname(url.fileURLToPath(import.meta.url)), 'foo.cjs')
+}
+
+function instrumentation(shim, resolvedModule) {
+  shim.wrapExport(resolvedModule, function wrapModule(shim, fn) {
+    return function wrappedModule() {
+      // `fn` _should_ be the `foo()` function exported by the module.
+      // If it is anything else, i.e. the proxy object, then we have an error
+      // in our handling of CJS modules as ESM.
+      const foo = fn.apply(this, arguments)
+      const _name = foo.name
+      foo.name = () => {
+        const value = _name.call(foo)
+        return `wrapped: ${value}`
+      }
+      return foo
+    }
+  })
+}
+
+tap.beforeEach(async (t) => {
+  shimmer.registerInstrumentation({
+    type: InstrumentationDescriptor.TYPE_GENERIC,
+    moduleName: 'foo',
+    absolutePath: modPath,
+    onRequire: instrumentation
+  })
+
+  const agent = helper.instrumentMockedAgent()
+  t.context.agent = agent
+
+  const { default: foo } = await import('./foo.cjs?v=' + crypto.randomBytes(16).toString('hex'))
+  t.context.mod = foo
```

Comment:
I don't think I understand the question. I have to import it after the "agent" is setup because otherwise the `registerHooks` will not pick up the module.

https://github.com/newrelic/node-newrelic/blob/cb21d2c95e5bb0de7e16535ecd4b2f5a77dc6fb7/test/lib/agent_helper.js#L180-L190

Reply:
oh so i meant importing the file to get at `default`, `foo` and just the raw module.exports.  Looking at this right now you're just getting default correct?